src/egon/data/datasets/power_plants/wind_farms.py

```diff
@@ -79,34 +79,49 @@ def insert():
     summary_t = pd.DataFrame()
     farms = pd.DataFrame()
 
-    # Fit wind farms scenarions for each one of the states
-    for scenario in target_power_df.index:
-        state_wf = gpd.clip(wf_areas, target_power_df.at[scenario, "geom"])
-        state_wf_ni = gpd.clip(
-            wf_areas_ni, target_power_df.at[scenario, "geom"]
-        )
-        state_mv_districts = gpd.clip(
-            mv_districts, target_power_df.at[scenario, "geom"]
-        )
-        target_power = target_power_df.at[scenario, "capacity"]
-        scenario_year = target_power_df.at[scenario, "scenario_name"]
-        source = target_power_df.at[scenario, "carrier"]
-        fed_state = target_power_df.at[scenario, "name"]
+    if "eGon100RE" in target_power_df["scenario_name"].values:
         wind_farms_state, summary_state = wind_power_states(
-            state_wf,
-            state_wf_ni,
-            state_mv_districts,
-            target_power,
-            scenario_year,
-            source,
-            fed_state,
+            wf_areas,
+            wf_areas_ni,
+            mv_districts,
+            target_power_df["capacity"].values[0],
+            "eGon100RE",
+            "wind_onshore",
+            "DE",
         )
-        summary_t = pd.concat([summary_t, summary_state])
-        farms = pd.concat([farms, wind_farms_state])
+        target_power_df = target_power_df[
+            target_power_df["scenario_name"] != "eGon100RE"
+        ]
+
+    if "eGon2035" in target_power_df["scenario_name"].values:
+        # Fit wind farms scenarions for each one of the states
+        for scenario in target_power_df.index:
```

Comment:
I am wondering if this loop is needed when eGon100RE was already created in the upper function calls.

Reply:
It is a process that is performed by federal state, so the for loop is necessary. I think the variable name is misleading, so I changed it in b3f87c8ff842201f20e90fe95bdba676ab060c13